### PR TITLE
Exclude gcp packages from dependabot

### PR DESCRIPTION
**Pull request by damccorm**
These packages are managed separately and shouldn't be auto-updated.

R: @suztomo 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Add a link to the appropriate issue in your description, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -27,3 +27,22 @@ updates:
...
+    - dependency-name: "grpc_protobuf_lite"
+    - dependency-name: "guava"
+    - dependency-name: "guava_testlib"
+    - dependency-name: "protobuf_java"
+    - dependency-name: "protobuf_java_util"
+    - dependency-name: "google_cloud_datastore_v1_proto_client"
+    - dependency-name: "google_cloud_pubsublite"
+    - dependency-name: "google_cloud_platform_libraries_bom"
+    - dependency-name: "proto_google_cloud_datastore_v1"
+    - dependency-name: "proto_google_cloud_pubsublite_v1"
```
Comment (suztomo) [inline .github/dependabot.yml:48]: These are keys in a map `project.ext.library` defined in BeamModulePlugin.groovy. Does this really work for DependaBot?

Comment (damccorm) [reply .github/dependabot.yml]: Oh you're right, I got my wires crossed - this should be fixed now.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -27,3 +27,17 @@ updates:
...
     schedule:
       interval: "daily"
+    ignore:
+    # Ignore gcp dependencies
+    - dependency-name: "com.google.api:gax*"
+    - dependency-name: "com.google.cloud.bigdataoss:*"
+    - dependency-name: "com.google.cloud:google-cloud-spanner*"
+    - dependency-name: "com.google.code.gson:*"
+    - dependency-name: "com.google.oauth-client:*"
+    - dependency-name: "io.grpc:grpc-protobuf*"
```
Comment (suztomo) [inline .github/dependabot.yml:37]: Can you add more gRPC dependencies?

```
        grpc_alts                                   : "io.grpc:grpc-alts", // google_cloud_platform_libraries_bom sets version
        grpc_api                                    : "io.grpc:grpc-api", // google_cloud_platform_libraries_bom sets version
        grpc_auth                                   : "io.grpc:grpc-auth", // google_cloud_platform_libraries_bom sets version
        // Once grpc-xds is added to google_cloud_platform_libraries_bom, use google_cloud_platform_libraries_bom instead.
        grpc_census                                 : "io.grpc:grpc-census:$grpc_version",
        grpc_context                                : "io.grpc:grpc-context", // google_cloud_platform_libraries_bom sets version
        grpc_core                                   : "io.grpc:grpc-core", // google_cloud_platform_libraries_bom sets version
```

File: .github/dependabot.yml [hunk 3]
```diff
@@ -27,3 +27,17 @@ updates:
     directory: "/" # Location of package manifests
     schedule:
       interval: "daily"
+    ignore:
+    # Ignore gcp dependencies
+    - dependency-name: "com.google.api:gax*"
+    - dependency-name: "com.google.cloud.bigdataoss:*"
+    - dependency-name: "com.google.cloud:google-cloud-spanner*"
```
Comment (suztomo) [inline .github/dependabot.yml:34]: How about `com.google.cloud:*`?

File: .github/dependabot.yml [hunk 4]
```diff
@@ -27,3 +27,17 @@ updates:
...
+    - dependency-name: "com.google.code.gson:*"
+    - dependency-name: "com.google.oauth-client:*"
```
Comment (suztomo) [inline .github/dependabot.yml:36]: Can you also add `com.google.http-client:*`?

File: .github/dependabot.yml [hunk 5]
```diff
@@ -27,3 +27,17 @@ updates:
...
+    - dependency-name: "io.grpc:grpc-protobuf*"
+    - dependency-name: "com.google.guava:*"
+    - dependency-name: "com.google.protobuf:*"
+    - dependency-name: "com.google.cloud.datastore:datastore-v1-proto-client"
+    - dependency-name: "com.google.cloud:google-cloud-pubsub*"
+    - dependency-name: "com.google.cloud:libraries-bom"
+    - dependency-name: "com.google.api.grpc:proto-google-cloud-*"
```
Comment (suztomo) [inline .github/dependabot.yml:43]: Can you also add `com.google.api.grpc:grpc-*` (Notice the difference "proto-" and "grpc-" prefix)

Can you also add `com.google.apis:google-api-services-*`?

File: .github/dependabot.yml (hunk 3 above)
Comment (damccorm) [reply .github/dependabot.yml]: Done!

File: .github/dependabot.yml (hunk 2 above)
Comment (damccorm) [reply .github/dependabot.yml]: Sure, updated to `io.grpc:grpc-*`

File: .github/dependabot.yml (hunk 4 above)
Comment (damccorm) [reply .github/dependabot.yml:36]: Done!

File: .github/dependabot.yml (hunk 5 above)
Comment (damccorm) [reply .github/dependabot.yml:41]: Done!

**Review by suztomo — APPROVED** [review]
Thank you!

**Comment by damccorm** [comment]
@suztomo could you please merge? (I'm not a committer)

**Pull request merged**